Habitica/src/main/java/com/habitrpg/android/habitica/ui/fragments/preferences/PreferencesFragment.java

```diff
@@ -197,6 +204,36 @@ public void onSharedPreferenceChanged(SharedPreferences sharedPreferences, Strin
             } else {
                 pushNotificationManager.removePushDeviceUsingStoredToken();
             }
+        } else if (key.equals("language")) {
+            LanguageHelper languageHelper = new LanguageHelper(sharedPreferences.getString(key,"en"));
+            Locale.setDefault(languageHelper.getLocale());
+            Configuration configuration = new Configuration();
+
+            if (android.os.Build.VERSION.SDK_INT <= Build.VERSION_CODES.JELLY_BEAN){
+                configuration.locale = languageHelper.getLocale();
+            } else {
+                configuration.setLocale(languageHelper.getLocale());
+            }
+
+            getActivity().getResources().updateConfiguration(configuration,
+                    getActivity().getResources().getDisplayMetrics());
```

Comment:
Can you add that it also reloads the content with the appropriate new language? Otherwise the content will not have the same language and would only change after a while, which would be a strange user experience.